src/Microsoft.DotNet.XHarness.InstrumentationBase.Xunit/AndroidInstrumentationBase.cs

```diff
@@ -0,0 +1,192 @@
+
+            if (File.Exists(entryPoint.TestsResultsFinalPath))
+            {
+                bundle.PutString(ResultsPathArgumentName, entryPoint.TestsResultsFinalPath);
+            }
+            
+            if (bundle.GetLong(ExitCodeArgumentName, 82) == (long)ExitCode.RETURN_CODE_NOT_SET)
+            {
+                bundle.PutLong(ExitCodeArgumentName, (long)ExitCode.RETURN_CODE_NOT_SET);
+            }
+
+            Finish(Result.Ok, bundle);
+        }
+
+        class TestsEntryPoint : AndroidApplicationEntryPoint
+        {
+            readonly string _resultsPath;
+            readonly string _excludeCategoriesDir;
+            readonly string _excludeCategoriesFile;
+            private Dictionary<string, string> _parsedArguments = new Dictionary<string, string>();
+
+            protected override string IgnoreFilesDirectory => _excludeCategoriesDir;
+            protected override string IgnoredTraitsFilePath => _excludeCategoriesFile;
+
+            public TestsEntryPoint(Bundle bundle)
+            {
+                var root = Application.Context.GetExternalFilesDir(null)?.AbsolutePath;
+
+                if (root == null)
+                {
+                    throw new InvalidOperationException("Unable to retrieve tests results final path.");
+                }
+
+                var docsDir = Path.Combine(root, "Documents");
+
+                if (!Directory.Exists(docsDir))
+                {
+                    Directory.CreateDirectory(docsDir);
+                }
+
+                if (bundle != null)
+                {
+                    foreach (var key in bundle.KeySet())
+                    {
+                        var value = bundle.GetString(key);
+                        if (!string.IsNullOrEmpty(value))
+                        {
+                            _parsedArguments.Add(key, value);
+                        }
+                    }
+                }
+
+                // use default name for test results file
+                _parsedArguments.TryAdd(ResultsFileArgumentName, "TestResults.xml");
+
+                _resultsPath = Path.Combine(docsDir, _parsedArguments[ResultsFileArgumentName]);
+                _excludeCategoriesDir = _parsedArguments[ExcludeCategoriesDirArgumentName];
+                _excludeCategoriesFile = _parsedArguments[ExcludeCategoriesFileArgumentName];
+            }
+
+            protected override bool LogExcludedTests => true;
+
+            public override TextWriter? Logger => null;
+
+            public override string TestsResultsFinalPath => _resultsPath;
+
+            protected override int? MaxParallelThreads => System.Environment.ProcessorCount/2;
+
+            protected override IDevice? Device => null;
+
+            public IEnumerable<Assembly> Tests
+            {
+                get
+                {
+                    return new List<Assembly>();
+                }
+                internal set { }
+            }
```

Comment:
This property looks very very strange. Lets thing about it for a second. 

get => Will always return a new List, if we for example do the following:

```csharp
var first = this.Tests;
var second = this.Tests;
Object.ReferenceEquals (first, first); // will return true
Object.ReferenceEquals (first, second); // will return false
```
Therefore we do have 2 instances that are different, either we do not want a property, we have an unusual behaviour or we got a bug.

Lets know look at the set. The way you wrote it is doing **nothing**. So 
```csharp
this.Test = new List<Assembly>(){"foo", "bar"}; // will have no effect, we alway return an empty list
var first = this.Test; // results in an empty list
```
You probably wanted to do the following:
```csharp
public IEnumerable<Assembly> Tests { get; set; } = new List<Assembly>();
```
The **get** and the **set** will be generated by the compiler, and it will be init to be an empty list. Take a look at https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/auto-implemented-properties

Reply:
fixed